drivers/flash/nrf_qspi_nor.c

```diff
@@ -482,12 +482,18 @@ static int qspi_rdsr(const struct device *dev, uint8_t sr_num)
 }
 
 /* Wait until RDSR confirms write is not in progress. */
-static int qspi_wait_while_writing(const struct device *dev, k_timeout_t poll_period)
+static int qspi_wait_while_writing(const struct device *dev, uint32_t poll_period_us)
 {
 	int rc;
 
 	do {
-		k_sleep(poll_period);
+		if (poll_period_us > 0) {
+#ifdef CONFIG_MULTITHREADING
+			k_sleep(K_USEC(poll_period_us));
+#else
+			k_busy_wait(poll_period_us);
```

Comment:
What is the point of busy waiting here? If there are no other contexts trying to execute (since there is no multithreading), why not just poll continuously in a loop like before?
```
#ifdef CONFIG_MULTITHREADING
     k_sleep(poll_period);
#endif
```

Reply:
TBH: I just copied the approach done in other places where this occurs.

Might be more power-saving to not poll it constantly for 10 millisecond though.

Reply:
It's either erasing or writing flash (the random winbond flash chip I looked in the datasheet for consumes 20mA in that state), I doubt what we do here matters, and it didn't matter for the last 5 years of the driver :)

Not sure where else you're seeing  the `k_busy_wait`, but I would suspects thats for delays that need to be respected for the chip to operate.

Reply:
The other instances of `k_busy_wait` was in mcuboot, places like this: https://github.com/zephyrproject-rtos/mcuboot/blob/main/boot/zephyr/io.c#L140

@anangl Do you agree with Jordan about removing the k_busy_wait?

Reply:
Agreed. I also don't see a clear benefit of this busy waiting between readouts of status register, so let's get rid of it.

Reply:
> The other instances of k_busy_wait was in mcuboot, places like this: https://github.com/zephyrproject-rtos/mcuboot/blob/main/boot/zephyr/io.c#L140

The reason was years ago, `k_sleep()` was not defined for non-multithreading builds, it was only January this year https://github.com/zephyrproject-rtos/zephyr/pull/63994 it was added

Reply:
> The reason was years ago, k_sleep() was not defined for non-multithreading builds, it was only January this year https://github.com/zephyrproject-rtos/zephyr/pull/63994 it was added

I didn't know that k_sleep now handled the case for non-multithreading builds now.
I guess this is not really a bug-fix then, but this patch was required in order to backport https://github.com/zephyrproject-rtos/zephyr/pull/75861 to NCS 2.6 and 2.7.
I think having this patch is still useful so that users can cherry-pick that patch and this one to fix QSPI if they are at those NCS version.

Without that patch these NCS versions have issues with QSPI in my application.
1. QSPI does not sleep and the power-consumption is quite high.
error:
`<err> qspi_nor: pm_device_runtime_put failed: -16`
2. Getting some file-system corruption during boot when going to a different version of the firmware:
```
<err> littlefs: WEST_TOPDIR/modules/fs/littlefs/lfs.c:1234: Corrupted dir pair at {0x1, 0x0}
<wrn> littlefs: can't mount (LFS -84); formatting
<inf> littlefs: /lfs1 mounted
<err> fs: file open error (-2)
```

Reply:
@anangl FYI: In case you run into these kinds of bug-reports.